src/back/hlsl/writer.rs

```diff
@@ -1856,14 +1856,66 @@ impl<'a, W: fmt::Write> super::Writer<'a, W> {
                     }
                 };
 
+                let pointer_is_atomic_scalar = {
+                    fn is_atomic_scalar_ty<'a>(
+                        module: &'a Module,
+                        func_ctx: &'a back::FunctionCtx,
+                        ty: Handle<crate::Type>,
+                    ) -> bool {
+                        match &module.types[ty].inner {
+                            TypeInner::Atomic {
+                                // TODO: floats and booleans aren't valid to interlock, so uh…problem?
+                                kind: crate::ScalarKind::Sint | crate::ScalarKind::Uint,
```

Comment:
todo: We actually _should_ be handling floats and booleans here, I'm guessing, but support doesn't seem straightforward to me in my (very limited) search:

* Floats need a specific shader model version or above, not sure which yet. I've seen mention in shader model 6 docs, but that might not be the minimum version.
* No idea about boolean, but I haven't seen it mentioned in any docs I've seen so far.

Reply:
`ScalarKind` is indeed too loose for this particular operation. We forbid types other than i32 and u32 [here](https://github.com/gfx-rs/naga/blob/52043be00489d6dedb08c36bb97d018d11cb1e13/src/valid/type.rs#L306-L309).

Reply:
Gotcha. Should be resolvable, then. I removed my `TODO` comment, and I think otherwise this code should be correct for the question I had.